akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -843,10 +850,10 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
       // Don't mark gossip state as seen while exiting is in progress, e.g.
       // shutting down singleton actors. This delays removal of the member until
       // the exiting tasks have been completed.
-      if (exitingTasksInProgress)
-        latestGossip = winningGossip
-      else
-        latestGossip = winningGossip seen selfUniqueAddress
+      latestGossip =
+        (if (exitingTasksInProgress) winningGossip
+        else winningGossip seen selfUniqueAddress)
+          .pruneTombstones(System.currentTimeMillis() - PruneGossipTombstonesAfter.toMillis)
```

Comment:
Doesn't matter but since it's not urgent to prune I think it makes sense to do it on convergence.

If there is no other changes the pruning in receiveGossip will not trigger. Might not matter but from a design perspective I think it's a task for the leader.